client/ayon_hiero/plugins/create/create_workfile.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+"""Creator plugin for creating workfiles."""
+import ayon_api
+
+from ayon_core.pipeline.create import CreatedInstance, AutoCreator
+
+from ayon_hiero.api import tags
+
+
+class CreateWorkfile(AutoCreator):
+    """Workfile auto-creator."""
+    settings_category = "hiero"
+
+    identifier = "io.ayon.creators.hiero.workfile"
+    label = "Workfile"
+    product_type = "workfile"
+    icon = "fa5.file"
+
+    default_variant = "Main"
+
+    @classmethod
+    def dump_instance_data(cls, data):
+        """ Dump instance data into AyonData project tag.
+
+        Args:
+            data (dict): The data to push to the project tag.
+        """
+        project_tag = tags.get_or_create_workfile_tag(create=True)
+
+        tag_data = {
+            "metadata": data,
+            "note": "AYON workfile data"
+        }
+        tags.update_tag(project_tag, tag_data)
+
+    def load_instance_data(cls):
+        """ Returns the data stored in AyonData project tag if any.
+
+        Returns:
+            dict. The project data.
+        """
+        project_tag = tags.get_or_create_workfile_tag()
+        if project_tag is None:
+            return {}
+
+        instance_data = tags.get_tag_data(project_tag)
+        return instance_data
+
+    def _create_new_instance(self):
+        """Create a new workfile instance.
+
+        Returns:
+            dict. The data of the instance to be created.
+        """
+        project_name = self.create_context.get_current_project_name()
+        folder_path = self.create_context.get_current_folder_path()
+        task_name = self.create_context.get_current_task_name()
+        host_name = self.create_context.host_name
+        variant = self.default_variant
+
```

Comment:
NOTE: You can use
```python
folder_entity = self.create_context.get_current_folder_entity()
task_entity = self.create_context.get_current_task_entity()
```

Reply:
I have used that instead, thanks for the tip !